proparse/src/main/java/org/prorefactor/proparse/antlr4/PreproEval.java

```diff
@@ -0,0 +1,612 @@
+  public Object visitLesserThan(LesserThanContext ctx) {
+    return compare(visit(ctx.expr(0)), visit(ctx.expr(1)), Compare.LT);
+  }
+
+  @Override
+  public Object visitLesserEquals(LesserEqualsContext ctx) {
+    return compare(visit(ctx.expr(0)), visit(ctx.expr(1)), Compare.LE);
+  }
+
+  @Override
+  public Object visitPlus(PlusContext ctx) {
+    return opPlus(visit(ctx.expr(0)), visit(ctx.expr(1)));
+  }
+
+  @Override
+  public Object visitMinus(MinusContext ctx) {
+    return opMinus(visit(ctx.expr(0)), visit(ctx.expr(1)));
+  }
+
+  @Override
+  public Object visitDivide(DivideContext ctx) {
+    return opDivide(visit(ctx.expr(0)), visit(ctx.expr(1)));
+  }
+
+  @Override
+  public Object visitModulo(ModuloContext ctx) {
+    Double m1 = getFloat(visit(ctx.expr(0))) + .5;
+    Double m2 = getFloat(visit(ctx.expr(1))) + .5;
+    return new Integer(m1.intValue() % m2.intValue());
+  }
+
+  @Override
+  public Object visitNot(NotContext ctx) {
+    return new Boolean(!getBool(visit(ctx.expr())));
+  }
+
+  @Override
+  public Object visitUnaryMinus(UnaryMinusContext ctx) {
+    Object o = visit(ctx.expr());
+    if (o instanceof Integer)
+      return (Integer) o * -1;
+    else
+      return (Float) o * -1;
+  }
+
+  // ****
+  // Atom
+  // ****
+
+  @Override
+  public Object visitNumber(NumberContext ctx) {
+    return getNumber(ctx.NUMBER().getText());
+  }
+
+  @Override
+  public Object visitQuotedString(QuotedStringContext ctx) {
+    return StringFuncs.qstringStrip(ctx.QSTRING().getText());
+  }
+
+  @Override
+  public Object visitTrueExpr(TrueExprContext ctx) {
+    return Boolean.TRUE;
+  }
+
+  @Override
+  public Object visitFalseExpr(FalseExprContext ctx) {
+    return Boolean.FALSE;
+  }
+
+  @Override
+  public Object visitExprInParen(ExprInParenContext ctx) {
+    return visit(ctx.expr());
+  }
+
+  @Override
+  public Object visitUnknownExpr(UnknownExprContext ctx) {
+    return null;
+  }
+
+  // *********
+  // Functions
+  // *********
+
+  @Override
+  public Object visitIntegerFunction(IntegerFunctionContext ctx) {
+    return new Integer(ctx.expr().getText());
+  }
+
+  @Override
+  public Object visitInt64Function(Int64FunctionContext ctx) {
+    return new Integer(ctx.expr().getText());
+  }
+
+  @Override
+  public Object visitProversionFunction(ProversionFunctionContext ctx) {
+    return "11.6";
+  }
+
+  /**
+   * Perfect :-)
+   * @see https://xkcd.com/221/
+   */
+  @Override
+  public Object visitRandomFunction(RandomFunctionContext ctx) {
+    return Integer.valueOf(4);
+  }
+
+  // *****************
+  // Support functions
+  // *****************
+
+  enum Compare {
+    EQ, NE, GT, GE, LT, LE
+  }
+
+  /**
+   * Test compare for two objects.
+   * 
+   * @see #compare(Object, Object)
+   */
+  static Boolean compare(Object left, Object right, Compare test) {
+    Integer result = compare(left, right);
+    if (result == null) {
+      if (test == Compare.NE)
+        return true;
+      return null;
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Null is returned but a "Boolean" is expected. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS2447)